jax/numpy/lax_numpy.py

```diff
@@ -1488,7 +1489,10 @@ def reduction(a, axis=None, dtype=None, out=None, keepdims=False):
     if out is not None:
       raise ValueError("reduction does not support the `out` argument.")
 
-    a = a if isinstance(a, ndarray) else asarray(a)
+
+    if isinstance(a, list):
+      raise TypeError(f"Reductions don't work on list, pass a ndarray, tuple or scalar instead")
```

Comment:
Can we move this test into a helper function? We should be consistent about what we accept in most of these NumPy functions, not just reductions, and that is probably most easily achieved with a helper function that does the checking, e.g., `_check_arguments_are_arrays_or_scalars(...)` or similar?

Do you think we should accept tuples only in reductions? I'd lean towards not accepting them anywhere, so I'm curious whether you thing the `jnp.prod(shape)` pattern is common.

Reply:
It caused many errors, so it seems to be common in testing at least, but I need to check the logs to see where exactly they were being used.

Reply:
I agree with @hawkinsp  that we should try to avoid both lists and tuples.

It might be cleaner to write a helper function like `_asarray_no_coercion()`, and just use that instead of `asarray()` internally.

Reply:
`host_callback_test`, `linalg_test` and `random_test` use this pattern, I could open a PR removing it, should I use `onp.prod` instead or do you guys prefer another fix?

Reply:
`onp.prod(x)` is a good alternative. So is `functools.reduce(operator, x, 1)`, if a little more verbose.

Reply:
I added the `_asarray_no_coercion()`, was this what you had in mind?